### PR TITLE
Added email field to bug report submission

### PR DESCRIPTION
**Pull request by vanessa-hu**
Based on Report Issue redesign at [client Issue 43!](https://github.com/keepid/keepid_client/issues/43)

- in BugReportMessage, new INVALID_EMAIL message
- in BugController's submitBug, now also retrieves email from JSON & returns INVALID_EMAIL error if not valid by ValidationUtils

### CODE REVIEW

**Comment by crchong1** [comment]
@vanessa-hu is this done?

**Comment by vanessa-hu** [comment]
@crchong1 server side should be done (just extra email check) but I guess the tests didn't pass?

**Comment by crchong1** [comment]
@vanessa-hu do you think you could actually take a look at the backend and modify the handler to use a Service? You can find examples of Services in the Services folder of each package.

**Comment by crchong1** [comment]
also @vanessa-hu i think you also included the username code in this PR too.

File: src/main/Bug/BugController.java [hunk 1]
```diff
@@ -35,8 +36,16 @@ public BugController(MongoDatabase db) {
         logger.info("Trying to get fields from form");
         JSONObject req = new JSONObject(ctx.body());
         JSONObject res = new JSONObject();
-        String title = req.getString("bugTitle");
-        String description = req.getString("bugDescription");
+        String title = req.getString("title");
```
Comment (crchong1) [inline src/main/Bug/BugController.java:39]: You should create a service class called SubmitIssueService (you can also rename bug to issue everywhere) which implements Service

File: src/main/Bug/BugReportMessage.java [hunk 2]
```diff
@@ -5,7 +5,8 @@
 public enum BugReportMessage {
```
Comment (crchong1) [inline src/main/Bug/BugReportMessage.java:5]: make BugReportMessage implement Message and then copy those fields/methods from UserMessage

File: src/main/User/UserController.java [hunk 3]
```diff
@@ -217,7 +217,24 @@ public Handler loginUser(SecurityUtils securityUtils, EmailUtil emailUtil) {
         ctx.json(res.put("username", candidateUsername).toString());
         logger.info("Username successfully generated");
       };
-
+  public Handler usernameExists =
```
Comment (crchong1) [inline src/main/User/UserController.java:220]: I think you left this in :) 
If you want though, if you write a service class for this, that would be great.

**Review by crchong1 — CHANGES_REQUESTED** [review]
This is a good addition to the backend. However, I would try to add more tests @vanessa-hu. With the Service layer, testing becomes a lot easier. If you want to mock the database, that might make it easy too.

File: src/main/Bug/BugReportMessage.java (hunk 2 above)
Comment (vanessa-hu) [reply src/main/Bug/BugReportMessage.java:5]: couldn't find Message interface (userMessage doesn't have it either)

**Comment by kofmangregory** [comment]
What is the status of this PR?

**Review by crchong1 — APPROVED** [review]

**Pull request merged**